src/shogun/base/SGObject.h

```diff
@@ -715,7 +721,10 @@ class CSGObject
 	void watch_param(
 		const std::string& name, T** value, S* rows, S* cols,
 		AnyParameterProperties properties = AnyParameterProperties(
-		    "Unknown parameter", MS_NOT_AVAILABLE, GRADIENT_NOT_AVAILABLE))
+			"Unknown parameter",
```

Comment:
Same with the description, I would offer a ctor that does not ask for one, and then set it to the empty string or "No description given" in the ctor, but not here in `SGObject.h`